IFTTT/ConnectSDK-iOS
# Handoff Connection activation to the IFTTT app when it is available

This is the SDK support for the in-app flow. 

**Refactors**
- Renames `Connection_internal.swift` to `Connection+parsing.swift`
- Removes `Connection+URLGeneration.swift`
- Moves email encoding from this file to `URLComponents+email.swift` and makes it an extension on `URLComponents` 
- Moves URL generation to a new type `ConnectionActivationFlow`

**What it changes**
- Adds an SDK setup requirement to set the scheme `ifttt-handoff-v1` in the partner app PLIST for `LSApplicationQueriesSchemes`
- Adds new `ActivationStep` for `appHandoff`

**How to Test**
Checkout this branch and https://github.com/IFTTT/GrizzlyiOS/pull/2396 and verify that you can handoff the connection flow from the SDK to the app.

- Verify that upon completing the flow, you are returned to the original app and the connection is enabled
- Verify that bailing on the flow and returning to the SDK cancels the connection

**TODO**
- [x] Document the new requirement to the README
- [x] Add a transition from `appHandoff` to `connected`. This is functionally the same as: https://ifttt-dev.atlassian.net/browse/IFTTT-1608. Let's leave it for this ticket. 

**Open Questions**
- There is an edge case that we need to define the behavior for: 

1. The user starts the handoff flow to the IFTTT app
2. The user cancels and returns to the original app
3. The user returns to the IFTTT app and completes the flow
This is documented here: https://ifttt-dev.atlassian.net/browse/IFTTT-1611. We can ignore in this PR.
